Reinstall option

NasKar2 opened the pull request:
Add option to reinstall to existing data.  Must set options DB_ROOT_PASSWORD=""
DB_PASSWORD="" in the config file.

basilhendroff reviewed (changes requested):
Hi NasKar,
Thanks for the suggested code changes. I have some comments:
1. Is it necessary to define the DB passwords in the config file. I can't see where in the code you use them?
2. Have you tested the reinstall option to see that it works?
3. What I also wrestled with for the detection of a new installation (around line 121) is when I first put together this code, whether the test is comprehensive enough e.g. the act of putting anything in those empty directories (e.g. touch test.txt) is enough to trigger an 'Oh, there's a valid installation here!'.

NasKar2 commented:
1. I removed the DB passwords from the config file.  Original had them there to correctly write them to the /root/wordpress_db_password.txt file.  When you do a reinstall of wordpress with the same data the passwords don't change so wordpress_db_password.txt is still correct.
2. I've test the reinstall and it does work
3. I changed to reinstall check to look for the presence of the file "${FILES_PATH}/wp-config.php".  If that file is there it should be ok to do a resintall.  If not they should wipe out the data directory and start from scratch.

basilhendroff commented:
Hi NasKar,

Getting closer. I'm beginning to see where you're heading with this. What I'm starting to recognise is that what's lacking is a clear definition of 'reinstall'. I've come to realise that what it means is that the data outside the jail is considered valid, but the jail itself needs to be rebuilt because, for instance, it's been destroyed.

So, in rebuilding the jail, the guiding principle should be that nothing outside the jail is touched. This would mean for instance that, in a reinstall, the code that configures WP should not be executed (it still is in your version), because those files live outside the jail.

I'm going to close this pull request, but work on something based on your ideas. The main reason is, I think you've exposed your db password in one of the commits. You may want to change your db password now.
